### PR TITLE
CRUD

### PR DESCRIPTION
**Pull request by cw323**
- Update README with API routes and instructions
- Create schema for PostgreSQL and Cassandra
- Integrate database methods into server file
- Made server directory file name change from server.js to index.js

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,19 +17,46 @@
 
 ## API
 ### URL
-> `http://localhost:3001/rooms/:id/` Dynamically render rooms based on listing id.
+> `http://localhost:3001/gallery/:listingId/` Dynamically render gallery based on listing id.
```
Comment (quetzalcaotl) [inline README.md]: Would this URL not make more sense as:
```suggestion
> `http://localhost:3001/:listingId/gallery` Dynamically render gallery based on listing id.
```

File: README.md [hunk 2]
```diff
@@ -17,19 +17,46 @@
...
+<!-- this is the url for the photo-gallery module -->
 
 ### POST
-> `/add/:id/` Add a photo in listing.
+> `/:listingId/upload` Add a photo into a specific listing.
+
+**Example**
+req body:
+  `{
+   photo_url: String
```
Comment (quetzalcaotl) [inline README.md]:
```suggestion
   photo_url: String,
```

File: README.md [hunk 3]
```diff
@@ -17,19 +17,46 @@
...
+   caption: String
+  }`
+
+res: 201
 
 ### GET
-> `/:id/:photoId` Retreive a photo in listing.
+> `/gallery/:listingId/photos` Retreive all photos in the gallery for a specific listing. (or do I want to get ALL of the photos)
```
Comment (quetzalcaotl) [inline README.md]: Same as above:
```suggestion
> `/:listingId/gallery/photos` Retrieve all photos in the gallery for a specific listing. (or do I want to get ALL of the photos)
```

File: README.md [hunk 4]
```diff
@@ -17,19 +17,46 @@
...
+<!--get all method querying the listings table-->
+
+**Example**
+`/gallery/1/photos` will return photo gallery for listingId: 1.
+`/gallery/8/photos` will return photo gallery for listingId: 8.
 
 ### PUT
-> `/update/:id/:photoId` Update a photo in listing.
+> `/gallery/:photoId` Update a photo caption or photo URL.
```
Comment (quetzalcaotl) [inline README.md]: Same as above:
```suggestion
> `/:listingId/gallery/:photoId` Update a photo caption or photo URL.
```

File: README.md [hunk 5]
```diff
@@ -17,19 +17,46 @@
...
+  `{
+    caption: String
+    photoUrl: String
+  }`
+
+res: 200
 
 ### DELETE
-> `/delete/:id/:photoId` Delete a photo in listing.
+> `/gallery/:photoId` Delete a photo in listing.
```
Comment (quetzalcaotl) [inline README.md]: Same as above:
```suggestion
> `/:listingId/gallery/:photoId` Delete a photo in listing.
```

File: README.md (hunk 1 above)
Comment (quetzalcaotl) [reply README.md]: It's fine if it doesn't, I just wanted to offer this as a consideration.

File: README.md [hunk 6]
```diff
@@ -17,19 +17,46 @@
...
+res: 201
 
 ### GET
-> `/:id/:photoId` Retreive a photo in listing.
+> `/gallery/:listingId/photos` Retreive all photos in the gallery for a specific listing. (or do I want to get ALL of the photos)
+<!--get all method querying the listings table-->
+
+**Example**
+`/gallery/1/photos` will return photo gallery for listingId: 1.
+`/gallery/8/photos` will return photo gallery for listingId: 8.
```
Comment (quetzalcaotl) [inline README.md]: You show the response on the other requests. I would consider showing what these results actually look like.

File: schema.cql [hunk 7]
```diff
@@ -0,0 +1,10 @@
+CREATE KEYSPACE IF NOT EXISTS 'photo_gallery';
```
Comment (quetzalcaotl) [inline schema.cql]: Does `'photo_gallery'` need to be in quotes?

File: schema.sql [hunk 8]
```diff
@@ -0,0 +1,14 @@
+DROP DATABASE IF EXISTS photo_gallery;
+CREATE DATABASE photo_gallery;
+
+CREATE TABLE listings (
+    listing_id SERIAL PRIMARY KEY
+);
```
Comment (quetzalcaotl) [inline schema.sql]: Maybe I just need an explanation, but I don't really understand the functionality of the listings table if all it contains is all possible listings. Wouldn't this also be contained within your photos table?

File: README.md (hunk 1 above)
Comment (cw323) [reply README.md]: Thank you for the suggestion, the page is rendered based on the listing id and not the gallery itself.

File: schema.sql [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+CREATE TABLE photos (
+    listing_id INTEGER NOT NULL,
+    photo_id SERIAL NOT NULL PRIMARY KEY, // BIG SERIAL?
```
Comment (quetzalcaotl) [inline schema.sql]: I'm not sure if the `//` style comments work in .sql documents. [PostgreSQL Docs on Comments](https://www.postgresql.org/docs/8.0/sql-syntax.html#SQL-SYNTAX-COMMENTS)

File: README.md (hunk 6 above)
Comment (cw323) [reply README.md]: I'll update the README once I test out the requests!

File: server/index.js [hunk 10]
```diff
@@ -13,10 +14,10 @@ app.use(function(req, res, next) {
 
 app.use('/', express.static(path.join(__dirname, '../public')));
 
-app.use('/rooms/:id', express.static(path.join(__dirname, '../public')));
+app.use('/gallery/:listingId', express.static(path.join(__dirname, '../public')));
```
Comment (quetzalcaotl) [inline server/index.js]: Same as previous comments:
```suggestion
app.use('/:listingId/gallery/', express.static(path.join(__dirname, '../public')));
```

File: schema.cql (hunk 7 above)
Comment (cw323) [reply schema.cql]: Made the update. Thanks!

File: README.md (hunk 6 above)
Comment (quetzalcaotl) [reply README.md]: Perfectly fine!

File: schema.sql (hunk 8 above)
Comment (cw323) [reply schema.sql]: The photos table references the listings table and is useful for future tables that will need to reference it, for example, user likes.

**Pull request merged**